database/landlord/2024_05_06_221309_data_fill_setup_complete_on_tenants_table.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Migrations\Migration;
+
+return new class () extends Migration {
+    public function up(): void
+    {
+        if (Schema::hasColumn('tenants', 'setup_complete')) {
+            DB::table('tenants')->update(['setup_complete' => true]);
+        }
+    }
+
+    public function down(): void
+    {
+        // nop
```

Comment:
@dgoetzit and/or @danharrin do y'all see an issue with there not being a `down` here?

I feel like we talked about these types of scenarios before but forget where we landed. Obviously it is relatively impossible to revert this, since we can't know which items were updated. But will having nothing in down affect rollbacks?

Reply:
Personally, I would prefer it if all data migrations had a `down()` method where possible, but I have previously been told that was not necessary so have not been doing it recently

Reply:
Yeah that does make sense. So because they all started out as false, it shouldn't hurt for the down here to just set all to false?

Reply:
The only reason it would hurt would be if a user had set it to true manually, but if we ever needed to revert then we'd probably have bigger problems

Reply:
True, I don't view the downs as a tool for disaster recovery. We would never use them in production. Only so we can rollback for migration testing.

So with that @mxm1070 if you can update the data migrations to include the `down`